crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -2187,38 +2146,84 @@ fn make_objects_history_to_commit(
+                    // b/c following deletion / mutation always has a higher version.
+                    // Technically, assertions below are not required, double check just in case.
+                    Either::Left((id, mutation)) => {
+                        let mutation_version = mutation.object.version();
+                        if let Some(existing) = deletions.remove(&id) {
+                            assert!(
+                                existing.object_version < mutation_version.value(),
+                                "Mutation version ({:?}) should be greater than existing deletion version ({:?}) for object {:?}",
+                                mutation_version,
+                                existing.object_version,
+                                id
+                            );
+                        }
+                        if let Some(existing) = mutations.insert(id, mutation) {
+                            assert!(
+                                existing.object.version() < mutation_version,
+                                "Mutation version ({:?}) should be greater than existing mutation version ({:?}) for object {:?}",
+                                mutation_version,
+                                existing.object.version(),
+                                id
+                            );
+                        }
+                    }
+                    Either::Right((id, deletion)) => {
+                        let deletion_version = deletion.object_version;
+                        if let Some(existing) = mutations.remove(&id) {
+                            assert!(
+                                existing.object.version().value() < deletion_version,
+                                "Deletion version ({:?}) should be greater than existing mutation version ({:?}) for object {:?}",
+                                deletion_version,
+                                existing.object.version(),
+                                id
+                            );
+                        }
+                        if let Some(existing) = deletions.insert(id, deletion) {
+                            assert!(
+                                existing.object_version < deletion_version,
+                                "Deletion version ({:?}) should be greater than existing deletion version ({:?}) for object {:?}",
+                                deletion_version,
+                                existing.object_version,
+                                id
+                            );
+                        }
+                    }
                 }
-            }
-        }
-    }
-
-    deleted_objects
-        .into_values()
-        .chain(latest_objects.into_values().map(StoredObjectSnapshot::from))
-        .collect()
+                (mutations, deletions)
+            },
+        );
+    (
+        mutations.into_values().collect(),
+        deletions.into_values().collect(),
```

Comment:
Probably simpler if we just turn them into the Stored format here?

Reply:
I kept them in indexed so that `objects` and `objects_snapshot` can share codes here